native/explorer/src/encoding.rs

```diff
@@ -324,18 +348,44 @@ pub fn list_from_series(data: ExSeries, env: Env) -> Term {
     let s = &data.resource.0;
 
     match s.dtype() {
-        DataType::Boolean => encode!(s, env, bool),
-        DataType::Int32 => encode!(s, env, i32),
-        DataType::Int64 => encode!(s, env, i64),
-        DataType::UInt8 => encode!(s, env, u8),
-        DataType::UInt32 => encode!(s, env, u32),
-        DataType::Utf8 => encode_utf8_series(&data.resource, s, env),
-        DataType::Float64 => encode_float64_series(s, env),
-        DataType::Date => encode_date_series(s, env),
-        DataType::Datetime(time_unit, None) => encode_datetime_series(s, *time_unit, env),
+        DataType::Boolean => series_to_list!(s, env, bool),
+        DataType::Int32 => series_to_list!(s, env, i32),
+        DataType::Int64 => series_to_list!(s, env, i64),
+        DataType::UInt8 => series_to_list!(s, env, u8),
+        DataType::UInt32 => series_to_list!(s, env, u32),
+        DataType::Utf8 => utf8_series_to_list(&data.resource, s, env),
+        DataType::Float64 => float64_series_to_list(s, env),
+        DataType::Date => date_series_to_list(s, env),
+        DataType::Datetime(time_unit, None) => datetime_series_to_list(s, *time_unit, env),
         DataType::List(t) if t as &DataType == &DataType::UInt32 => {
-            encode_list!(s, env, u32, u32)
+            list_series_to_list!(s, env, u32, u32)
         }
         dt => panic!("to_list/1 not implemented for {:?}", dt),
     }
 }
+
+#[allow(clippy::size_of_in_element_count)]
+pub fn iovec_from_series(data: ExSeries, env: Env) -> Term {
+    let s = &data.resource.0;
+    let resource = &data.resource;
+
+    match s.dtype() {
+        DataType::Boolean => {
+            let mut bin = OwnedBinary::new(s.len()).unwrap();
+            let slice = bin.as_mut_slice();
+            for (i, v) in s.bool().unwrap().into_iter().enumerate() {
+                slice[i] = v.unwrap() as u8;
+            }
+            [bin.release(env)].encode(env)
+        }
+        DataType::Int32 => series_to_iovec!(resource, s, env, i32, i32),
```

Comment:
Would be possible to avoid passing both `resource` and `s`? I don't have a strong opinion about this, but since you can refer to the resource from the series, maybe you can simplify this. WDYT?

Reply:
I tried but then I need to pass the data as a reference or similar and I got stuck :D

Reply:
Let's go with this and I will try later to remove the duplication (it's easier with the LSP :P).